index.js

```diff
@@ -299,6 +299,8 @@ function solveChallenge (options, response, body) {
       cause = 'Failed to parse challenge timeout';
       return callback(new errors.ParserError(cause, options, response));
     }
+  } else {
+    console.warn('Warning: `cloudflareTimeout` is deprecated. Cloudscraper automatically handles timeout for you. Use `cloudflareMaxTimeout` to reduce timeout if you need to');
```

Comment:
@pro-src question for you. Should we silence `console` calls in test runs to remove this from the output? I could create an abstraction above logger and properly stub it in the tests, but not for a single call :)

Reply:
It will be easier to encapsulate this method and disable in tests :/

Reply:
`cloudflareTimeout` is still useful but not as a default option. You can use it to specify the timeout which is probably a more common use case than overriding `cloudflareMaxTimeout`. If you use `cloudflareTimeout`, cloudscraper will still work if parsing the timeout fails. It's still being used in tests. The only real change there was that it was removed `cloudscraper.defaultParams` so cloudscraper will default to parsing the timeout.

Reply:
Deprecated the option only serves as a very verbose way of saying that you don't really need this option anymore. It was always an optional setting and unless they're setting in below 4 seconds then there is no need to modify their code. If they were setting it below 4 seconds, they were having problems anyway... :laughing:

Reply:
Do you plan on removing it and/or renaming `cloudflareMaxTimeout` to `cloudflareTimeout` or something?

Reply:
@pro-src actually yes, I do want to get rid off `cloudflareTimeout` in the next major release, as with your changes library basically hides the process of challenge solving and secondary things like waiting for a required time and etc., so why to have `cloudflareTimeout` together with `cloudflareMaxTimeout` ?

Reply:
I mentioned how the option was still useful and hinted at the fact that one is not an alias of the other. Although, you can achieve a similar effect by setting a very low maximum timeout. Another reason why it's useful is that setting a longer timeout might give better results in some cases or at least aid in troubleshooting. `cloudflare-scrape` had mentioned something along those lines in their README or was it a message in their source??? Point being, there are some use cases to consider. This gives more control to the user and we can make similar arguments against `cloudflareMaxTimeout`. If you wanted, you could look at it as being a min and max. In any case, I would expect `cloudflareTimeout` to be used more frequently than `cloudflareMaxTimeout` since it has sensible default. If you don't like having both, wouldn't it be preferable to remove `cloudflareMaxTimeout`?

Reply:
I thought you might want to make `cloudflareTimeout` only ever advisory unless parsing the timeout failed or would result in a longer timeout. An easy way of achieving that would be to rename `cloudflareMaxTimeout` to `cloudflareTimeout` but then you lose the ability to say ***always use this specific timeout***. My comments about the frequency of option use is purely speculation. I haven't polled the user base. It's probably worth mentioning that the timeout until recently was 6 seconds when it only needed to be 4 so you never know... A lot of users could be using the option right now.

Reply:
I hope my thoughts are helpful and not annoying. I'm happy with however you proceed with this. Just glad to be contributing :smile:

Reply:
Thank you for the explanations :) I will then leave both and make a comment in the readme about when one option should be used with or without other.